File: jpp/lab2i3/java/DHTest.java
public class DHTest {
    public static void main(String[] args) {
        GF.setP(1234567891);
        FieldOperations<GF> op = new GFOperations(); 
        DHSetup<GF, FieldOperations<GF>> setup = new DHSetup<>(op);

        User<GF, FieldOperations<GF>> alice = new User<>(setup,op);
        User<GF, FieldOperations<GF>> bob = new User<>(setup,op);
        System.out.println("Alice's public key: " + alice.getPublicKey());
        System.out.println("Bob's public key: " + bob.getPublicKey());

        alice.setKey(bob.getPublicKey());
        bob.setKey(alice.getPublicKey());

       
        GF message = new GF(123456);
        GF encrypted = alice.encrypt(message);
        GF decrypted = bob.decrypt(encrypted);

       
        System.out.println("Message: " + message);
        System.out.println("Encrypted: " + encrypted);
        
        System.out.println("Decrypted: " + decrypted);
        
        message = new GF(456461);
        encrypted = alice.encrypt(message);
        decrypted = bob.decrypt(encrypted);

        
        System.out.println("Message: " + message);
        System.out.println("Encrypted: " + encrypted);
        System.out.println("Decrypted: " + decrypted);
        
    }
}

File: jpp/lab2i3/cpp/DHtest.cpp
#include "User.hpp"
#include "DHSetup.hpp"
#include "Gf.hpp"
int main()
{
    Gf::setP(1234567891);
    DHSetup<Gf> setup;
    User<Gf> alice(setup);
    User<Gf> bob(setup);

    std::cout << "Alice's public key: " << alice.getPublicKey() << std::endl;
    std::cout << "Bob's public key: " << bob.getPublicKey() << std::endl;

    alice.setKey(bob.getPublicKey());
    bob.setKey(alice.getPublicKey());

    Gf message = 123456;
    Gf encrypted = alice.encrypt(message);
    Gf decrypted = bob.decrypt(encrypted);

    std::cout << "Message: " << message << std::endl;
    std::cout << "Encrypted: " << encrypted << std::endl;
    std::cout << "Decrypted: " << decrypted << std::endl;

    message = 456461;
    encrypted = alice.encrypt(message);
    decrypted = bob.decrypt(encrypted);

    std::cout << "Message: " << message << std::endl;
    std::cout << "Encrypted: " << encrypted << std::endl;
    std::cout << "Decrypted: " << decrypted << std::endl;

    return 0;
}